File: Project/src/app/models/flight.ts
export class Flight {
    id:number;
    takeOffLocation:string;
    landingLocation:string;
    takeOffDate:string;
    landingDate:string;
    takeOffTime:string;
    landingTime:string;
    duration:number;
    numberOfChangeovers:number;
    changeoverLocations:Array<string>;
    price:number;
    rate:number;
    image:string;
    economySeats:number;
    premiumEconomySeats:number;
    businessSeats:number;
    firstClassSeats:number;
    airline:string;
    

  constructor(id:number,takeOffLocation:string, landingLocation:string, takeOffDate:string,landingDate:string,
                takeOffTime:string,landingTime:string, duration:number, numberOfChangeovers:number,changeoverLocations:Array<string>,
                  price:number, rate:number,economySeats:number, premiumEconomySeats:number,businessSeats:number,
                  firstClassSeats:number, airline:string){

    this.id = id;
    this.takeOffLocation=takeOffLocation;
    this.landingLocation=landingLocation;
    this.takeOffDate=takeOffDate;
    this.landingDate=landingDate;
    this.takeOffTime=takeOffTime;
    this.landingTime=landingTime;
    this.duration=duration;
    this.numberOfChangeovers=numberOfChangeovers;
    this.changeoverLocations=changeoverLocations;
    this.price=price;
    this.rate=rate;
    this.economySeats=economySeats;
    this.premiumEconomySeats=premiumEconomySeats;
    this.businessSeats=businessSeats;
    this.firstClassSeats=firstClassSeats;
    this.airline=airline;
    
  }
}

File: Server/Server/Models/Flight.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Server.Models
{
    public class Flight
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string PlaneId { get; set; }

        [ForeignKey("PlaneId")]
        public virtual Plane Plane { get; set; }

        public Airport TakeOffLocation { get; set; }
        public Airport LandingLocation { get; set; }

        public string TakeOffDate { get; set; }
        public string LandingDate { get; set; }

        public string TakeOffTime{ get; set; }
        public string LandingTime { get; set; }

        public double Duration { get; set; }

        public ICollection<Airport> Connections { get; set; }
        public ICollection<SegmentPrice> SegmentPrices { get; set; }
        //public ICollection<FlightReservation> Reservations { get; set; }

        public ICollection<Seat> OccupiedSeats { get; set; }
    }
}
